editor/src/components/canvas/remix/utopia-remix-root-component.tsx

```diff
@@ -150,25 +153,45 @@ function useGetRoutes() {
       return routes
     }
 
+    const customServerCreator = remixDerivedDataRef.current.customServerCreator
     const creators = remixDerivedDataRef.current.routeModuleCreators
 
     function addExportsToRoutes(innerRoutes: DataRouteObject[]) {
       innerRoutes.forEach((route) => {
-        // FIXME Adding a loader function to the 'root' route causes the `createShouldRevalidate` to fail, because
-        // we only ever pass in an empty object for the `routeModules` and never mutate it
         const creatorForRoute = creators[route.id] ?? null
         if (creatorForRoute != null) {
           for (const routeExport of RouteExportsForRouteObject) {
-            route[routeExport] = (args: any) =>
-              creatorForRoute
-                .executionScopeCreator(
-                  projectContentsRef.current,
-                  fileBlobsRef.current,
-                  hiddenInstancesRef.current,
-                  displayNoneInstancesRef.current,
-                  metadataContext,
-                )
-                .scope[routeExport]?.(args) ?? null
+            route[routeExport] = async (args: any) => {
+              const { context: requestContext } = await getContextFromCustomServer(
+                customServerCreator,
+                projectContentsRef.current,
+                fileBlobsRef.current,
+                hiddenInstancesRef.current,
+                displayNoneInstancesRef.current,
+                metadataContext,
+                args.request,
+              )
+
+              const patchedArgs = {
```

Comment:
I found `patchRoutesWithContext` in @remix/testing and they only patch loader and action 

https://github.com/remix-run/remix/blob/4a8f558d16b8739d7f70903958ac5792e050f3e9/packages/remix-testing/create-remix-stub.tsx#L55